### PR TITLE
Hb sleep repo tests

### PR DESCRIPTION
**Pull request by H-Bo214**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

### Detailed Description
* Refactored tests for sleep-repo 
* Added tests for data-repo
* Removed .only's from other tests

### Files modified:
- [x] sleep-repo-tests.js
- [x] data-repo-tests.js

### CODE REVIEW

File: src/Sleep-repo.js [hunk 1]
```diff
@@ -84,12 +84,14 @@ class SleepRepo extends DataRepo {
     });
   }
 
-  getUsersWithMostSleepForDay(dataSet, userRepo) {
-    return dataSet.reduce((usersWithMaxSleep, data) => {
+  getUsersWithMostSleepForDay(dataSet, userRepo, date) {
+    dataSet = this.getMaxSleepData(date)
```
Comment (rwilliams659) [inline src/Sleep-repo.js:88]: If you do want to hard-code in the dataSet used in this method to be the return value from this.getMaxSleepData, then we no longer need the parameter of "dataSet" here. Alternatively, you can revert the method back to how it was, then to properly test this method in the test suite, you would just call getUsersWithMostSleepForDay, passing in this.getMaxSleepData(date) as the first parameter.

File: test/data-repo-test.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  it('should be a function', function() {
+    expect(DataRepo).to.be.a('function');
+  });
+
+  it('should be an instance of DataRepo', function() {
+    expect(dataRepo).to.be.an.instanceof(DataRepo);
+  });
+
+  it('should calculate an average', function() {
+    expect(Number(dataRepo.calculateAverage(sleepData,'sleepQuality').toFixed(1))).to.equal(2.6)
```
Comment (rwilliams659) [inline test/data-repo-test.js:81]: I think the current return value of calculateAverage is already a number, so I don't think that Number() is necessary, but I could be wrong! Just think we should double-check.

File: test/sleep-repo-test.js [hunk 3]
```diff
@@ -120,70 +161,68 @@ describe('Sleep', function() {
...
+
+  // This method will return multiple user results. Functioning correctly.
+
+it('should get users who slept the most number of hours on a specific date', function() {
+  expect(sleepRepo.getMaxSleepData('2019/08/22')).to.deep.eql([sleepData[0]])
+})
 
   it('should return people with longest sleep for the day', function() {
-    expect(sleep.getUsersWithMostSleepForDay('2019/06/21', userRepo)).to.eql(['Allie McCarthy', 'Rainbow Dash']);
+    expect(sleepRepo.getUsersWithMostSleepForDay(sleepData, userRepo, '2019/06/21')).to.eql([ 'Rainbow Dash', 'Allie McCarthy']);
```
Comment (rwilliams659) [inline test/sleep-repo-test.js:225]: See note from Sleep repo above!

File: test/data-repo-test.js (hunk 2 above)
Comment (H-Bo214) [reply test/data-repo-test.js:81]: This does return a number but it ends up being a long decimal number.  When I try to toFixed it it turns it into a string.  I'm not sure how to get rid of the trailing decimal number without using toFixed which converts the number into a string.

File: src/Sleep-repo.js (hunk 1 above)
Comment (H-Bo214) [reply src/Sleep-repo.js:88]: These changes have been implemented. Thanks!

File: test/data-repo-test.js (hunk 2 above)
Comment (H-Bo214) [reply test/data-repo-test.js:81]: These changes have been implemented. Thanks!

File: test/sleep-repo-test.js (hunk 3 above)
Comment (H-Bo214) [reply test/sleep-repo-test.js]: These changes have been implemented. Thanks!

**Comment by jkwest-93** [comment]
Really nice work with this testing suite!

**Comment by krogowsk531** [comment]
Great job with the tests!

**Comment by rwilliams659** [comment]
Looks perfect now!

**Pull request merged**